### PR TITLE
[oneDNN v3.x] Update fused instance norm kernel to support oneDNN v3.x

### PR DESCRIPTION
**Pull request by yimeisun123**
This PR adds support for oneDNN v3.x in fused instance norm kernel.
This PR passes all unit tests related to fused instance norm when oneDNN v3.x is enabled.
This PR does not add or affect Eigen ops and is specific only to oneDNN kernel.

### CODE REVIEW

**Comment by yimeisun123** [comment]
I checked the logs, the failed test case is  
MLIR tests :: replace_cast_hacks_with_tf_xla_ops_large_constants.mlir
the changes in this PR doesn't touch code related to the above test case.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay!

File: tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc [hunk 1]
```diff
@@ -221,23 +270,35 @@ class MklFusedInstanceNormOp : public OpKernel {
     return valid;
   }
 
-  // Helper function to add scale and shift data into same buffer in float
-  // type as requested by oneDNN
-  void SetupScaleShiftBuffer(const Tensor& scale_tensor,
+  // Helper function to prepare scale and shift data in float type as
+  // required by oneDNN library. Prior to oneDNN 3.x version, the library
+  // requires the final scale and shift data to be passed in the same buffer
+  // wherase the 3.x version requires separate buffers for scale and shift
```
[the remark this answers is not in the document]
Comment (yimeisun123) [reply tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc]: Updated, thank you!

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Pull request merged**